src/__utils/Collapse.tsx

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+export type CollapseProps = React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>;
+
+export const Collapse: React.FC<CollapseProps> = React.memo((props: CollapseProps) => {
+    const collapseRef: React.MutableRefObject<HTMLDivElement> = React.useRef<HTMLDivElement>();
+    const [height, setHeight] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        if (collapseRef && props["data-toggle"] !== undefined) {
+            setHeight(JSON.parse(props["data-toggle"]) ? collapseRef.current.scrollHeight : 0);
```

Comment:
Its dangerous to parse only after checking for undefined, either
```typescript
            try{
              setHeight(JSON.parse(props["data-toggle"]) ? collapseRef.current.scrollHeight : 0);
            }
            catch(){
              setHeight(0);
            }
```
or add more checks in the condition above

```typescript
if (collapseRef && props["data-toggle"]) {
```

Reply:
updated